src/konserve/impl/default.cljc

```diff
@@ -0,0 +1,495 @@
+(ns konserve.impl.default
+  "Default implementation of the high level protocol given a binary backing implementation as defined in the storage-layout namespace."
+  (:require
+   [konserve.serializers :refer [key->serializer]]
+   [konserve.compressor :refer [null-compressor]]
+   [konserve.encryptor :refer [null-encryptor]]
+   [hasch.core :refer [uuid]]
+   [clojure.string :refer [ends-with?]]
+   [konserve.protocols :refer [PEDNAsyncKeyValueStore -exists?
+                               PBinaryAsyncKeyValueStore
+                               -serialize -deserialize
+                               PKeyIterable]]
+   [konserve.impl.storage-layout :refer [-atomic-move -create-store
+                                         -copy -create-blob -delete -exists
+                                         -keys -path -sync-store
+                                         -close -get-lock -sync
+                                         -read-header -read-meta -read-value -read-binary
+                                         -write-header -write-meta -write-value -write-binary
+                                         PBackingLock -release
+                                         default-version
+                                         parse-header create-header]]
+   [konserve.utils :refer [async+sync *default-sync-translation*]]
+   [superv.async :refer [go-try- <?-]]
+   [clojure.core.async :refer [<! timeout]]
+   [taoensso.timbre :as timbre :refer [trace]])
+  #?(:clj
+     (:import
+      [java.io ByteArrayOutputStream ByteArrayInputStream])))
+
+(extend-protocol PBackingLock
+  nil
+  (-release [this env]
+    (if (:sync? env) nil (go-try- nil))))
+
+(defn update-blob
+  "Write file into file system. It write first the meta-size, that is stored in (1Byte),
+  the meta-data and the actual data."
```

Comment:
*writes

Reply:
Just to be clear, these docs were not written by me I just transported them through the refactoring. Actually it is even wrong, because this code is now not file system specific anymore. It is fixed now.